### PR TITLE
Orchestrator logs its own requests

### LINKED ISSUE #14252: [Theia AI] The orchestrator chat agent should log its own requests (138)
The delegating chat agent should log its own requests and responses in the history view. At the moment it's replicating the same request/response as the one it delegated to.

### PR DESCRIPTION
**Pull request by JonasHelming**
- add data field to request
- add option to turn off default logging
- extracted getJsonOfText
- add docu for getTextOfResponse

fixed #14252

#### What it does

Orchestrator logs its own requests separatly from the agent it delegates to.

It also adds a data field to 'request' to remember the orchestrator request id which is manually generated.

It refactors getJsonOfResponse and adds getJsonOfText

It adds documentation to getTextOfResponse to avoid other people falling into this trap: " **Important:** For stream responses, the stream can only be consumed once. Calling this function multiple times on the same stream response will return an empty string (`''`) on subsequent calls, as the stream will have already been consumed."

#### How to test

Do any chat requests and check the history view. For the orchestrator it should log the user requests and then the choosen agent. For the choosen agent, it should log the user request and they actual answer.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat/src/common/orchestrator-chat-agent.ts [hunk 1]
```diff
@@ -88,8 +90,19 @@ export class OrchestratorChatAgent extends AbstractStreamParsingChatAgent implem
     @inject(ChatAgentService)
     protected chatAgentService: ChatAgentService;
 
-    override invoke(request: ChatRequestModelImpl): Promise<void> {
+    override async invoke(request: ChatRequestModelImpl): Promise<void> {
         request.response.addProgressMessage({ content: 'Determining the most appropriate agent', status: 'inProgress' });
+        // We generate a dedicated ID for recording the orchestrator request/response, as we will forward the original request to another agent
+        const orchestartorRequestID = generateUuid();
```
Comment (planger) [inline packages/ai-chat/src/common/orchestrator-chat-agent.ts:96]:
```suggestion
        const orchestartorRequestId = generateUuid();
```

File: packages/ai-chat/src/common/orchestrator-chat-agent.ts [hunk 2]
```diff
@@ -100,8 +113,20 @@ export class OrchestratorChatAgent extends AbstractStreamParsingChatAgent implem
 
     protected override async addContentsToResponse(response: LanguageModelResponse, request: ChatRequestModelImpl): Promise<void> {
         let agentIds: string[] = [];
+        const responseText = await getTextOfResponse(response);
+        // We use the previously generated, dedicated ID to log the orchestrator response before we forward the original request
+        const orchestratorRequestID = request.getDataByKey(OrchestatorRequestIDKey);
```
Comment (planger) [inline packages/ai-chat/src/common/orchestrator-chat-agent.ts:118]:
```suggestion
        const orchestratorRequestId = request.getDataByKey(OrchestatorRequestIdKey);
```

File: packages/ai-chat/src/common/orchestrator-chat-agent.ts [hunk 3]
```diff
@@ -100,8 +113,20 @@ export class OrchestratorChatAgent extends AbstractStreamParsingChatAgent implem
...
+        if (typeof orchestratorRequestID === 'string') {
```
Comment (planger) [inline packages/ai-chat/src/common/orchestrator-chat-agent.ts:119]:
```suggestion
        if (typeof orchestratorRequestId === 'string') {
```

File: packages/ai-chat/src/common/orchestrator-chat-agent.ts [hunk 4]
```diff
@@ -100,8 +113,20 @@ export class OrchestratorChatAgent extends AbstractStreamParsingChatAgent implem
...
+            this.recordingService.recordResponse({
+                agentId: this.id,
+                sessionId: request.session.id,
+                timestamp: Date.now(),
+                requestId: orchestratorRequestID,
```
Comment (planger) [inline packages/ai-chat/src/common/orchestrator-chat-agent.ts:124]:
```suggestion
                requestId: orchestratorRequestId,
```

File: packages/ai-chat/src/common/chat-model.ts [hunk 5]
```diff
@@ -342,14 +343,29 @@ export class ChatRequestModelImpl implements ChatRequestModel {
...
         this._agentId = agentId;
+        this._data = data;
+    }
+
+    get data(): { [key: string]: unknown } | undefined {
+        return this._data;
+    }
+
+    addData(key: string, blob: unknown): void {
+        this._data[key] = blob;
```
Comment (planger) [inline packages/ai-chat/src/common/chat-model.ts:364]:
```suggestion
    addData(key: string, value: unknown): void {
        this._data[key] = value;
```

`value` might be a better name than `blob`, as it is likely no binary large object. But this is just a nit.

**Review by planger — APPROVED** [review]
Excellent, works great and code looks good. Inline just a few minor nitpicks. Thank you!

File: packages/ai-chat/src/common/orchestrator-chat-agent.ts [hunk 6]
```diff
@@ -88,8 +90,19 @@ export class OrchestratorChatAgent extends AbstractStreamParsingChatAgent implem
...
         request.response.addProgressMessage({ content: 'Determining the most appropriate agent', status: 'inProgress' });
+        // We generate a dedicated ID for recording the orchestrator request/response, as we will forward the original request to another agent
+        const orchestartorRequestId = generateUuid();
+        request.addData(OrchestatorRequestIdKey, orchestartorRequestId);
+        const userPrompt = request.request.text;
+        this.recordingService.recordRequest({
+            agentId: this.id,
+            sessionId: request.session.id,
+            timestamp: Date.now(),
+            requestId: orchestartorRequestID,
```
Comment (JonasHelming) [inline packages/ai-chat/src/common/orchestrator-chat-agent.ts:103]:
```suggestion
            requestId: orchestartorRequestId,
```

**Pull request merged**